### PR TITLE
docs(sda): fix typos and edit wording

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary
Resolves https://github.com/dagster-io/dagster/issues/7138.

Did a readthrough.

- remove i.e. and e.g.
- fix random typos

## Test Plan
vercel

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7136** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7136" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: docs/content/guides/dagster/software-defined-assets.mdx [hunk 1]
```diff
@@ -7,14 +7,16 @@ description: The "software-defined asset" APIs sit atop of the graph/job/op APIs
...
+In Dagster, an "asset" is a data product, an object produced by a data pipeline. Some examples are tables, machine learning models, or reports.
+
+Conceptually, software-defined assets invert the typical relationship between assets and computation. Instead of defining a graph of ops and recording which assets those ops end up materializing, you define a set of assets. Each asset knows how to compute its contents from upstream assets.
 
 Taking a software-defined asset approach has a few main benefits:
 
-- **Write less code** - because each asset knows about the assets it depends on, you don't need to use `@graph` / `@job` to wire up dependencies between your ops.
-- **Track cross-job dependencies via asset lineage** - Dagit allows you to find the parents and children of any asset, even if they live in different jobs. This is useful for finding the sources of problems and for understanding the consequences of changing or removing an asset.
+- **Write less code** - Each asset knows about the assets it depends on; you don't need to use `@graph` / `@job` to wire up dependencies between your ops.
```
Comment (jamiedemaria) [inline docs/content/guides/dagster/software-defined-assets.mdx:18]: should "you don't need to use `@graph` / `@job` to wire up dependencies between your ops" be "you don't need to use `@graph` / `@job` to wire up dependencies between your assets"?

Comment (sryza) [reply docs/content/guides/dagster/software-defined-assets.mdx:18]: I think maybe just leaving it "to wire up dependencies." would be best?

**Review by sryza — APPROVED** [review]
Thanks for catching these issues Rex.

**Pull request merged**